x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/connector/set_native_connector_api_logic.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ConnectorStatus } from '../../../../../common/types/connectors';
+import { createApiLogic } from '../../../shared/api_logic/create_api_logic';
+import { HttpLogic } from '../../../shared/http';
+import { NativeConnector } from '../../components/search_index/connector/types';
+
+export interface SetNativeConnectorArgs {
+  connectorId: string;
+  nativeConnector: NativeConnector;
+}
+
+export interface SetNativeConnectorResponse {
+  connectorId: string;
+  nativeConnector: NativeConnector;
+}
+
+export const setNativeConnector = async ({
```

Comment:
This is not a must, but I'd really like to move all of this into one backend call instead of three sequential ones that could potentially screw up in the frontend. It's easier to fix that in the backend.